chips/stm32f4xx/src/gpio.rs

```diff
@@ -1026,16 +1139,18 @@ impl hil::gpio::Pin for Pin<'a> {
             });
         }
     }
-}
 
-impl hil::gpio::PinCtl for Pin<'a> {
-    fn set_input_mode(&self, mode: hil::gpio::InputMode) {
-        match mode {
-            hil::gpio::InputMode::PullUp => self.set_pullup_pulldown(PullUpPullDown::PullUp),
-            hil::gpio::InputMode::PullDown => self.set_pullup_pulldown(PullUpPullDown::PullDown),
-            hil::gpio::InputMode::PullNone => {
-                self.set_pullup_pulldown(PullUpPullDown::NoPullUpPullDown)
-            }
+    fn set_client(&self, client: &'static hil::gpio::Client) {
+        self.client.set(client);
+    }
+
+    fn is_pending(&self) -> bool {
+        unsafe {
+            let val = self.exti_lineid.map(|lineid| {
+                let l = lineid.clone();
+                exti::EXTI.is_pending(l)
```

Comment:
`lineid` is copy dereference instead.

Also you could use map_or instead of `map + unwrap_or`

Reply:
Yes, much better!